.github/workflows/test.yml

```diff
@@ -19,4 +19,48 @@ jobs:
       uses: actions/checkout@v2
 
     - name: Run Tests
-      run: go test ./...
+      run: go test ./packages/...
+
+  integration-test:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Build GoShimmer Docker network
+      run: docker-compose -f tools/integration-tests/docker-compose.yml up -d --scale peer_replica=5
+
+    - name: Dispay containers
+      run: docker ps -a
+
+    - name: Run integration tests
+      run: docker-compose -f tools/integration-tests/tester/docker-compose.yml up --exit-code-from tester
+
+    - name: Stop GoShimmer Docker network
+      if: always()
+      run: docker-compose -f tools/integration-tests/docker-compose.yml stop
+
+    - name: Create logs from containers in network
+      if: always()
+      run: |
+        docker logs entry_node > tools/integration-tests/logs/entry_node.log
+        docker logs peer_master > tools/integration-tests/logs/peer_master.log
+        docker logs integration-tests_peer_replica_1 > tools/integration-tests/logs/peer_replica_1.log
+        docker logs integration-tests_peer_replica_2 > tools/integration-tests/logs/peer_replica_2.log
+        docker logs integration-tests_peer_replica_3 > tools/integration-tests/logs/peer_replica_3.log
+        docker logs integration-tests_peer_replica_4 > tools/integration-tests/logs/peer_replica_4.log
+        docker logs integration-tests_peer_replica_5 > tools/integration-tests/logs/peer_replica_5.log
+        docker logs tester > tools/integration-tests/logs/tester.log
+
+    - name: Save logs as artifacts
+      if: always()
+      uses: actions/upload-artifact@v1
+      with:
+        name: container-logs
+        path: tools/integration-tests/logs
+
+    - name: Clean up
+      if: always()
+      run: docker-compose -f tools/integration-tests/docker-compose.yml down --rmi all --remove-orphan --volumes
```

Comment:
I am not a big fan of `--rmi`. What is the purpose of this?
In an action it does not matter whatsoever, and in the context of `run-tests.sh` should not a `... up --build` have the same effect that we always have an updated image?